### PR TITLE
Edit user profile

### PR DESCRIPTION
**Pull request by gunnarrunner**
- Changes Implemented:
  - Created a pre-populated edit page for a current user.
  - Also have passing tests that cover that a user has been updated.
  - Tests pass travis

- Quality Control Checklist:

  - [x] Code adheres to Rubocop styling (if unavoidable infractions, please clarify below)
  - [ ] 100% SimpleCov test coverage (if below, please clarify below)
  - [x] Last commit has passing Travis build

- Blockers (if applicable):

- Next Steps & Additional Notes:

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -22,8 +22,24 @@ def dashboard
...
+    update_user = BackEndFacade.update_user(users_params, current_user.id)
+    # require "pry"; binding.pry
+    if update_user[:errors].present?
+      flash[:error] = "Could not update your profile please try again!"
+      redirect_to profile_edit_path(current_user.id)
+    else
+      flash[:success] = "Successfully updated your profile!"
+      redirect_to profile_path(current_user.id)
+    end
     # TODO: write this method and create view
```
Comment (tvaroglu) [inline app/controllers/users_controller.rb:37]: Nice use of the errors to guide the conditional!  Can you please drop the comments below the method

File: app/services/back_end_service.rb [hunk 2]
```diff
@@ -28,6 +28,15 @@ def create_user(user_params)
       )
     end
 
+    def update_user(user_params, user_id)
```
Comment (tvaroglu) [inline app/services/back_end_service.rb:31]: Did we confirm that this works?  (i.e. running two servers)?   Took a few tries to get this working on POST requests

File: spec/facades/back_end_facade_spec.rb [hunk 3]
```diff
@@ -12,7 +12,7 @@
     gyms = BackEndFacade.get_user_gyms(1)
 
     expect(gyms[0]).to be_an_instance_of(UserGym)
-    expect(gyms.count).to eq(4)
+    expect(gyms.count).to eq(3)
```
Comment (tvaroglu) [inline spec/facades/back_end_facade_spec.rb:15]: We need to be careful merging this... I cleaned this up, don't want this to get merged back in to hard-coded test values.

File: spec/features/users/edit_spec.rb [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+    allow(BackEndService).to receive(:update_user).and_return(JSON.parse(json_response, symbolize_names: true))
+
+    click_button("Update Profile")
+
+    expect(current_path).to eq("/profile/#{@user.id}")
+    expect(page).to have_content("Successfully updated your profile!")
+  end
+
+  it 'can fill out a form to update a user and gives a flash message when you successfully update a user' do
```
Comment (tvaroglu) [inline spec/features/users/edit_spec.rb:33]: Love the edge case testing!  Can you please update the name of the test in 33, so it's a bit more clear that this is our sad path?

File: spec/features/users/edit_spec.rb [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    expect(page).to have_field('Zip Code', with: "#{@user.zip_code}")
+    expect(page).to have_field('Summary', with: "#{@user.summary}")
+    expect(page).to have_field('Goals', with: "#{@user.goal}")
+
+    fill_in("Summary", with: "")
+
+    error_response = {                        
+                      :message=>"your query could not be completed",
+                      :errors=>[
+                      "Couldn't find User with 'id'=40"
```
Comment (tvaroglu) [inline spec/features/users/edit_spec.rb:45]: Line 45 should be `Summary can't be blank`

File: Gemfile.lock [hunk 6]
```diff
@@ -271,9 +271,8 @@ GEM
       rubocop (>= 1.7.0, < 2.0)
     rubocop-rake (0.6.0)
       rubocop (~> 1.0)
-    rubocop-rspec (2.4.0)
-      rubocop (~> 1.0)
-      rubocop-ast (>= 1.1.0)
+    rubocop-rspec (2.5.0)
+      rubocop (~> 1.19)
```
Comment (Scott-Borecki) [inline Gemfile.lock:275]: Oh boy!  RuboCop is getting stronger!!! 😱

File: app/controllers/users_controller.rb [hunk 7]
```diff
@@ -22,8 +22,24 @@ def dashboard
 
   def profile; end
 
+  def edit;  end
+  
   def update
+    update_user = BackEndFacade.update_user(users_params, current_user.id)
+    # require "pry"; binding.pry
+    if update_user[:errors].present?
```
Comment (Scott-Borecki) [inline app/controllers/users_controller.rb:30]: Looks good!  You can remove this pry as well.

**Comment by Scott-Borecki** [comment]
I know this was a challenging User Story, but I think you did a great job!  There are a couple of our comments to address and then this should be good to merge 🙌

File: Gemfile.lock (hunk 6 above)
Comment (tvaroglu) [reply Gemfile.lock:275]: XD
